src/grid/hip/grid_hip_context.h

```diff
@@ -33,50 +33,80 @@ class smem_parameters;
 template <typename T> class gpu_vector {
   size_t allocated_size_{0};
   size_t current_size_{0};
-  T *ptr = nullptr;
+  bool allocated_outside_{false};
+  T *device_ptr_ = nullptr;
+  T *host_ptr_ = nullptr;
 
 public:
-  gpu_vector() {
-    ptr = nullptr;
-    allocated_size_ = 0;
-    current_size_ = 0;
-  }
+  gpu_vector() {}
+
   // size is the number of elements not the memory size
-  gpu_vector(const size_t size_) {
-    if (size_ < 16) {
+  gpu_vector(const size_t size__) {
+    if (size__ < 16) {
       allocated_size_ = 16;
     } else {
-      allocated_size_ = (size_ / 16 + 1) * 16;
+      allocated_size_ = (size__ / 16 + 1) * 16;
     }
-    current_size_ = size_;
+    current_size_ = size__;
 
-    offloadMalloc((void **)&ptr, sizeof(T) * allocated_size_);
+#ifndef CP2K_USE_UNIFIED_MEMORY
+    offloadMalloc((void **)&device_ptr_, sizeof(T) * allocated_size_);
+#else
+    hipMallocManaged((void **)&device_ptr_, sizeof(T) * allocated_size_);
```

Comment:
What's the advantage of calling `hipMallocManaged` instead `hipMalloc` ?

Reply:
hipMalloc allocates on GPU only which means memory is not visible on the CPU side.